token-lending/program/src/processor.rs

```diff
@@ -1714,6 +1719,85 @@ fn unpack_mint(data: &[u8]) -> Result<Mint, LendingError> {
     Mint::unpack(data).map_err(|_| LendingError::InvalidTokenMint)
 }
 
+fn get_pyth_product_quote_currency(pyth_product: &pyth::Product) -> [u8; 32] {
+    let mut quote_currency = [0u8; 32];
+
+    let mut attribute_iter = pyth_product.attr[..].iter();
+    let mut attributes_size = pyth::PROD_ATTR_SIZE;
+
+    while attributes_size > 0 {
+        let mut size = *attribute_iter.next().unwrap() as usize;
+        let mut key = Vec::with_capacity(size);
+        for _ in 0..size {
+            key.push(*attribute_iter.next().unwrap());
+        }
+
+        if &key[..] == b"quote_currency" {
+            size = *attribute_iter.next().unwrap() as usize;
+            #[allow(clippy::needless_range_loop)]
+            for i in 0..size {
+                quote_currency[i] = *attribute_iter.next().unwrap();
+            }
+            break;
+        }
+
+        size = *attribute_iter.next().unwrap() as usize;
+        for _ in 0..size {
+            attribute_iter.next();
+        }
+
+        attributes_size -= 2 + key.len() + size;
+    }
+
+    quote_currency
+}
+
+fn get_pyth_price(pyth_price_info: &AccountInfo, clock: &Clock) -> Result<Decimal, ProgramError> {
+    const STALE_AFTER_SLOTS_ELAPSED: u64 = 5;
+
+    let pyth_price_data = pyth_price_info.try_borrow_data()?;
+    let pyth_price = pyth::load::<pyth::Price>(&pyth_price_data)
+        .map_err(|_| ProgramError::InvalidAccountData)?;
+
+    if pyth_price.ptype as u32 != pyth::PriceType::Price as u32 {
```

Comment:
nit: you can derive `PartialEq` for `PriceType` and then check for equality without casting

Reply:
Done.

Reply:
Done.